radixdlt/src/main/java/com/radixdlt/statecomputer/CommandToBinaryConverter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.statecomputer;
+
+import com.radixdlt.serialization.DsonOutput;
+import com.radixdlt.serialization.Serialization;
+import com.radixdlt.serialization.SerializationException;
+import com.radixdlt.syncer.CommittedCommand;
+
+public final class CommandToBinaryConverter {
+	private final Serialization serializer;
+
+	public CommandToBinaryConverter(Serialization serializer) {
+		this.serializer = serializer;
+	}
+
+	public byte[] toLedgerEntryContent(CommittedCommand command) {
+		try {
+			return serializer.toDson(command, DsonOutput.Output.PERSIST);
+		} catch (SerializationException e) {
+			throw new RuntimeException(String.format("Serialization for Command %s failed", command));
```

Comment:
Really should be an IllegalArgumentException

Reply:
not sure, it could be a bad serializer which is throwing the exception